packages/lesswrong/components/messaging/ConversationContents.tsx

```diff
@@ -0,0 +1,169 @@
+import stringify from "json-stringify-deterministic";
+
+const styles = (theme: ThemeType): JssStyles => ({
+  conversationTitle: {
+    ...theme.typography.commentStyle,
+    marginTop: 8,
+    marginBottom: 12,
+  },
+  editor: {
+    margin: '32px 0px',
+    position: "relative",
+    '& .form-submit': {
+      // form-submit has display: block by default, which for some reason makes it take up 0 height
+      // on mobile. This fixes that.
+      display: "flex",
+    }
+  },
+  backButton: {
+    color: theme.palette.lwTertiary.main,
+  },
+  row: {
+    display: "flex",
+    justifyContent: "space-between",
+  },
+});
+
+const ConversationContents = ({
+  conversation,
+  currentUser,
+  scrollRef,
+  classes,
+}: {
+  conversation: ConversationsList;
+  currentUser: UsersCurrent;
+  scrollRef?: React.RefObject<HTMLDivElement>;
+  classes: ClassesType;
+}) => {
+  // Count messages sent, and use it to set a distinct value for `key` on `NewMessageForm`
+  // that increments with each message. This is a way of clearing the form, which works
+  // around problems inside the editor related to debounce timers and autosave and whatnot,
+  // by guaranteeing that it's a fresh set of react components each time.
+  const [messageSentCount, setMessageSentCount] = useState(0);
+
+  const stateSignatureRef = useRef(stringify({conversationId: conversation._id, numMessagesShown: 0}));
+
+  const {
+    results,
+    refetch,
+    loading,
+  } = useMulti({
+    terms: {
+      view: "messagesConversation",
+      conversationId: conversation._id,
+    },
+    collectionName: "Messages",
+    fragmentName: "messageListFragment",
+    fetchPolicy: "cache-and-network",
+    limit: 100000,
+    enableTotal: false,
+  });
+
+  const { query } = useLocation();
+  const { captureEvent } = useTracking();
+
+  // Whenever either the number of messages changes, or the conversationId changes,
+  // scroll to the bottom. This happens on pageload, and also happens when the messages
+  // list is refreshed because of the useOnNotificationsChanged() call below, if the refresh
+  // increased the message count.
+  //
+  // Note, if you're refreshing (as opposed to navigating or opening a new
+  // tab), this can wind up fighting with the browser's scroll restoration (see
+  // client/scrollRestoration.ts).
+  useEffect(() => {
+    const newNumMessages = results?.length ?? 0;
+    const newStateSignature = stringify({conversationId: conversation._id, numMessagesShown: newNumMessages});
+    if (newStateSignature !== stateSignatureRef.current) {
+      stateSignatureRef.current = newStateSignature;
+      setTimeout(() => {
+        // Always scroll the whole window. This may be a problem in future, but it's here to make
+        // scroll work nicely on both desktop (uses inner div) and mobile (uses whole window)
+        window.scroll({top: document.body.scrollHeight-550, behavior: 'smooth'})
```

Comment:
What's this `550` constant? Is it just eyeballed or is it the height of something? If the later it should probably be defined as a constant somewhere and imported.